python/lsst/pipe/tasks/loadReferenceCatalog.py

```diff
@@ -0,0 +1,380 @@
+    )
+
+    def validate(self):
+        super().validate()
+        if self.doApplyColorTerms and len(self.colorterms.data) == 0:
+            msg = "applyColorTerms=True requires the `colorterms` field be set to a ColortermLibrary."
+            raise pexConfig.FieldValidationError(LoadReferenceCatalogConfig.colorterms, self, msg)
+
+
+class LoadReferenceCatalogTask(pipeBase.Task):
+    """Load multi-band reference objects from a reference catalog.
+
+    Parameters
+    ----------
+    dataIds : iterable of `lsst.daf.butler.dataId`, optional
+        An iterable object of dataIds which point to reference catalogs
+        in a Gen3 repository.  Required for Gen3.
+    refCats : iterable of `lsst.daf.butler.DeferredDatasetHandle`, optional
+        An iterable object of dataset refs for reference catalogs in
+        a Gen3 repository.  Required for Gen3.
+    butler : `lsst.daf.persistence.Butler`, optional
+        A Gen2 butler.  Required for Gen2.
+    """
+    ConfigClass = LoadReferenceCatalogConfig
+    _DefaultName = "loadReferenceCatalog"
+
+    def __init__(self, dataIds=None, refCats=None, butler=None, **kwargs):
+        if dataIds is not None and refCats is not None:
+            pipeBase.Task.__init__(self, **kwargs)
+            refConfig = self.config.refObjLoader
+            self.refObjLoader = ReferenceObjectLoader(dataIds=dataIds,
+                                                      refCats=refCats,
+                                                      config=refConfig,
+                                                      log=self.log)
+        elif butler is not None:
+            self.makeSubtask('refObjLoader', butler=butler)
+        else:
+            raise RuntimeError("Must instantiate LoadReferenceCatalogTask with "
+                               "dataIds/refCats (Gen3) or butler (Gen2)")
+
+        if self.config.doReferenceSelection:
+            self.makeSubtask('referenceSelector')
+        self._fluxFilters = None
+        self._fluxFields = None
+        self._referenceFilter = None
+
+    def getPixelBoxCatalog(self, bbox, wcs, filterList, epoch=None,
+                           bboxToSpherePadding=None):
+        """Get a multi-band reference catalog by specifying a bounding box and WCS.
+
+        The catalog will be in `numpy.ndarray`, with positions proper-motion corrected to
+        "epoch" (if specified, and if the reference catalog has proper motions);
+        sources cut on a reference selector (if
+        "config.doReferenceSelection = True"); and color-terms applied (if
+        "config.doApplyColorTerms = True").
+
+        The format of the reference catalog will be of the format:
+
+        dtype = [('ra', 'np.float64'),
+                 ('dec', 'np.float64'),
+                 ('refMag', 'np.float32', (len(filterList), )),
+                 ('refMagErr', 'np.float32', (len(filterList), ))]
+
+        Reference magnitudes (AB) will be 99 for non-detections in a given band.
```

Comment:
As per the comment above, perhaps adding "and errors" here (and elsewhere this is mentioned) could be useful.